src/ui/components/SuggestedResults/index.js

```diff
@@ -0,0 +1,54 @@
+/* @flow */
+/* eslint react/no-array-index-key: 0 */
+import React from 'react';
+import escapeStringRegexp from 'escape-string-regexp';
+
+import './style.scss';
+
+type Props = {
+  query: string,
+  suggestions: Array<string>,
+  onSelect: Function,
+};
+
+const renderSuggestion = (query: string, suggestion: string, onSelect: Function) => {
+  let parts;
+  if (query.replace(/\s+/, '') === '') {
+    parts = [suggestion];
+  } else {
+    parts = suggestion.split(new RegExp(`(${escapeStringRegexp(query)})`, 'gi'));
```

Comment:
Sorry, I'm not immediately understanding what this does. Could you explain/add some documentation to it?